File: lc/linkedlist/java/RemoveListElements.java
package lc.linkedlist.java;

public class RemoveListElements {
    public ListNode removeElements(ListNode head, int value) {
        ListNode dummyHead = new ListNode(0);
        dummyHead.next = head;
        ListNode iterator = dummyHead;

        while(head != null) {
            if(head.val == value) {
                iterator.next = head.next;
            } else {
                iterator = iterator.next;
            }
            head = head.next;
        }

        return dummyHead.next;
    }
}

File: lc/linkedlist/ruby/remove_list_elements.rb
def remove_elements(head, value)
    dummy_head = ListNode.new 0
    dummy_head.next = head
    iterator = dummy_head

    while head
        if head.val == value
            iterator.next = head.next
        else
            iterator = iterator.next
        end
        head = head.next
    end

    dummy_head.next
end